1_data/get_data_spice2.py

```diff
@@ -0,0 +1,287 @@
+"""Functions to obtain and process SPICE2 data for the workflow.
+
+This module provides functionality to download, process, and filter SPICE2 dataset
+for molecular property prediction workflows. It includes functions to:
+
+1. Download SPICE2 HDF5 data from Zenodo
+2. Process the raw HDF5 data into a HuggingFace structured dataset
+
+The module can be used as a command-line script or imported as a library.
+
+Based on: https://github.com/fjclark/descent-workflow/blob/main/workflow/get_data.py
+
+Command-line Usage
+------------------
+python get_data_spice2.py --data-dir PATH
+
+Command-line Arguments
+----------------------
+--data-dir : str
+    Directory path for SPICE2 data storage and processing. The directory
+    will be created if it doesn't exist. This is where the SPICE-2.0.1.hdf5
+    file will be downloaded and where processed datasets will be saved.
+
+Examples
+--------
+# Download and process SPICE2 data to local directory
+python get_data_spice2.py --data-dir ./spice_data
+
+# Process data to specific path
+python get_data_spice2.py --data-dir /path/to/data/directory
+
+Outputs
+-------
+raw-spice : dir
+├── dataset_info.json          # Dataset metadata and schema
+├── state.json                 # Dataset state information
+└── data-00000-of-00001.arrow  # Actual data in Apache Arrow format
+    ├── smiles (str)
+    ├── coords (list) # flattened, reshape with np.array(coords_list).reshape(-1, 3)
+    ├── energy (list) # flattened
+    └── forces (list) # flattened
+
+Constants
+---------
+HARTREE_TO_KCAL : float
+    Conversion factor from Hartree to kcal/mol
+BOHR_TO_ANGSTROM : float
+    Conversion factor from Bohr to Angstrom
+SPICE2_SOURCES : Set[str]
+    Set of SPICE2 dataset sources to include (excludes B/Si sets)
+"""
+
+from __future__ import annotations
+
+import argparse
+import json
+import pathlib
+import subprocess
+from typing import Any, Dict, List, Set
```

Comment:
Similarly native Python typing has been around for a while, I think since 3.10? So instead of e.g. `typing.Set[str]`, you should be able to natively use `set[str]` without imports.